DigitalLearningSolutions.Web/Views/LearningPortal/Current/SetCompleteByDate.cshtml

```diff
@@ -1,96 +1,54 @@
-          </span>
-        }
-        @if (Model.CompleteByValidationResult?.DateValid == false) {
-          <span class="nhsuk-error-message" id="validation-message" role="alert">
-            <span class="nhsuk-u-visually-hidden">Error:</span>@Model.CompleteByValidationResult?.ErrorMessage
-          </span>
-        }
-        <div class="nhsuk-date-input" id="date">
-          <div class="nhsuk-date-input__item">
-            <div class="nhsuk-form-group">
-              <label class="nhsuk-label nhsuk-date-input__label" for="day">
-                Day
-              </label>
-              <input class="nhsuk-input nhsuk-date-input__input nhsuk-input--width-2 @dayErrorClass" id="day" name="day" type="number" value="@prefillDay">
-            </div>
-          </div>
-          <div class="nhsuk-date-input__item">
-            <div class="nhsuk-form-group">
-              <label class="nhsuk-label nhsuk-date-input__label" for="month">
-                Month
-              </label>
-              <input class="nhsuk-input nhsuk-date-input__input nhsuk-input--width-2 @monthErrorClass" id="month" name="month" type="number" value="@prefillMonth">
-            </div>
-          </div>
-          <div class="nhsuk-date-input__item">
-            <div class="nhsuk-form-group">
-              <label class="nhsuk-label nhsuk-date-input__label" for="year">
-                Year
-              </label>
-              <input class="nhsuk-input nhsuk-date-input__input nhsuk-input--width-4 @yearErrorClass" id="year" name="year" type="number" value="@prefillYear">
-            </div>
-          </div>
-        </div>
-        @if (Model is CurrentCourseViewModel) {
-          <input name="progressId" type="hidden" value="@Model.ProgressId" />
-        } else {
-          <input name="selfAssessmentId" type="hidden" value="@Model.Id" />
-        }
-      </fieldset>
-      <button class="nhsuk-button nhsuk-u-margin-top-5" id="save-button" type="submit">Save changes</button>
+<div class="nhsuk-grid-row">
+  <div class="nhsuk-grid-column-full">
+    @if (errorHasOccurred) {
+      <vc:error-summary order-of-property-names="@(new[] { nameof(Model.Day), nameof(Model.Month), nameof(Model.Year) })" />
+    }
+
+    <h1 class="nhsuk-heading-xl word-break">Enter a completion date for @Model.Name</h1>
+
+    <form asp-action="@action"
+          class="nhsuk-u-margin-bottom-3"
+          method="post"
+          novalidate>
+      <input type="hidden" asp-for="Name" />
+      <input type="hidden" asp-for="Type" />
+      @if (Equals(Model.Type, LearningItemType.Course)) {
+        <input name="progressId" type="hidden" value="@Model.ProgressId" />
+      }
+      <vc:date-input id="complete-by-date"
+                     label="Complete by date"
+                     day-id="Day"
+                     month-id="Month"
+                     year-id="Year"
+                     css-class=""
+                     hint-text-lines="@hintTextLines" />

```

Comment:
This doesn't look to be passing in any values for the day/month/year. We probably want to so any previously set complete by dates will be displayed correctly.

Reply:
Does the date input take values like that? From other examples I didn't see any, and it seems to pre-populate the existing complete by date just fine as is

Reply:
Ah, I'd forgotten how the date input vc works, it does some fancy property name grabbing from the day-id etc. values. Should be fine.